### PR TITLE
chore(web): cleanup promise handling

### PR DESCRIPTION
**Pull request by benmccann**
These are a subset of the errors pointed out by eslint when setting [recommended-type-checked](https://typescript-eslint.io/users/configs#recommended-type-checked).

### CODE REVIEW

File: web/src/lib/utils.ts [hunk 1]
```diff
@@ -194,3 +194,9 @@ export const findLocale = (code: string | undefined) => {
     name: language?.name,
   };
 };
+
+export const asyncTimeout = (ms: number) => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+};
```
Comment (mertalev) [inline web/src/lib/utils.ts:202]: There's a promise version of `setTimeout` you can import from `timers/promises`

File: web/src/routes/admin/jobs-status/+page.svelte [hunk 2]
```diff
@@ -4,28 +4,31 @@
...
     jobs = await getAllJobsStatus();
   };
 
+  let running = true;
+
   onMount(async () => {
-    await load();
-    timer = setInterval(load, 5000);
+    while (running) {
+      await load();
```
Comment (mertalev) [inline web/src/routes/admin/jobs-status/+page.svelte:25]: There doesn't seem to be a point in having this `load` function over `jobs = await getAllJobsStatus();`

**Review by mertalev — COMMENTED** [review]
Looks really good! I'm surprised there are so many places where we don't await

File: web/src/routes/admin/jobs-status/+page.svelte (hunk 2 above)
Comment (benmccann) [reply web/src/routes/admin/jobs-status/+page.svelte]: yeah, good catch. updated

File: web/src/lib/utils.ts (hunk 1 above)
Comment (benmccann) [reply web/src/lib/utils.ts:202]: Ah, thanks! I didn't know about that one. Can we import Node built-ins on the client? I'm not sure whether Vite will bundle Node built-ins or not. I'll have to test later when I get a minute

Comment (mertalev) [reply web/src/lib/utils.ts:202]: Oops, I think not. I get `[plugin:vite:resolve] Module "timers/promises" has been externalized for browser compatibility` followed by `RollupError: [vite-plugin-sveltekit-compile] "setTimeout" is not exported by "__vite-browser-external"`.

**Review by mertalev — APPROVED** [review]

File: web/src/routes/admin/jobs-status/+page.svelte [hunk 3]
```diff
@@ -4,28 +4,27 @@
...
-  const load = async () => {
-    jobs = await getAllJobsStatus();
-  };
+  let running = true;
 
   onMount(async () => {
-    await load();
-    timer = setInterval(load, 5000);
+    while (running) {
+      await getAllJobsStatus();
```
Comment (michelheusschen) [inline web/src/routes/admin/jobs-status/+page.svelte:21]: Jobs don't show up on the page
```suggestion
      jobs = await getAllJobsStatus();
```

File: web/src/routes/(user)/albums/[albumId]/+page.svelte [hunk 4]
```diff
@@ -220,12 +220,12 @@
 
   onMount(async () => {
     if (album.sharedUsers.length > 0) {
-      getFavorite();
-      getNumberOfComments();
+      await getFavorite();
+      await getNumberOfComments();
```
Comment (michelheusschen) [inline web/src/routes/(user)/albums/[albumId]/+page.svelte:224]: This now runs sequentially instead of in parallel, but it probably doesn't matter that much

File: web/src/lib/utils.ts [hunk 5]
```diff
@@ -194,3 +195,13 @@ export const findLocale = (code: string | undefined) => {
...
+
+export const resolvePromise = <T>(promise: Promise<T>): void => {
+  Promise.resolve(promise).catch(async (error) => await handleError(error));
+};
```
Comment (michelheusschen) [inline web/src/lib/utils.ts:207]: Why is `Promise.resolve` needed when only promises are accepted as input? Also I don't think the name `resolvePromise` properly communicates that the error will get handled.

**Review by michelheusschen — COMMENTED** [review]
Will it still be possible to call an async function in `onMount` and (without awaiting) return a cleanup function? For example to abort a request.

File: web/src/lib/stores/assets.store.ts [hunk 6]
```diff
@@ -175,7 +175,7 @@ export class AssetStore {
     for (const bucket of this.buckets) {
       if (height < viewport.height) {
         height += bucket.bucketHeight;
-        this.loadBucket(bucket.bucketDate, BucketPosition.Visible);
+        await this.loadBucket(bucket.bucketDate, BucketPosition.Visible);
```
Comment (michelheusschen) [inline web/src/lib/stores/assets.store.ts:178]: Will cause buckets to load sequentially and might slow down initial loading a bit

File: web/src/routes/(user)/albums/[albumId]/+page.svelte (hunk 4 above)
Comment (benmccann) [reply web/src/routes/(user)/albums/[albumId]/+page.svelte]: updated

File: web/src/lib/utils.ts (hunk 5 above)
Comment (jrasm91) [reply web/src/lib/utils.ts]: All this does it automatically notify the user when an error happens. Maybe something like this would be better `notifyOnError(() => loadFavorites())`

Comment (benmccann) [reply web/src/lib/utils.ts]: updated

File: web/src/lib/components/faces-page/assign-face-side-panel.svelte [hunk 7]
```diff
@@ -128,7 +128,7 @@
       searchedPeopleCopy = data;
       searchWord = searchName;
     } catch (error) {
-      handleError(error, "Can't search people");
+      await handleError(error, "Can't search people");
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/assign-face-side-panel.svelte:131]: Would it make sense to just not await all of these, but change `handleError` to return void instead?

File: web/src/routes/(user)/trash/+page.svelte [hunk 8]
```diff
@@ -24,10 +24,15 @@
...
+  import { handlePromiseError } from '$lib/utils';
 
   export let data: PageData;
 
-  $: $featureFlags.trash || goto(AppRoute.PHOTOS);
+  featureFlags.subscribe(({ trash }) => {
+    if (!trash) {
+      handlePromiseError(goto(AppRoute.PHOTOS));
+    }
+  });
```
Comment (jrasm91) [inline web/src/routes/(user)/trash/+page.svelte:35]: Does this create memory leaks? I don't see any unsubscribing happening here.

This begs a question though, in onMount hooks you can use async/await to be compliant with the rule, but that doesn't mean the implementation can't throw an promise rejection. It assume `onMount` (svelte) will handle it.

The same thing is true here. If you throw an error on `goto` svelte will just handle it. If the goal is to avoid throwing unhandled promise rejections to the runtime there are other things we should be changing, not just updating stuff to be "compliant" with the rule. 

Is there a reason you didn't just use `$featureFlags.trash || handlePromiseError(goto(AppRoute.Photos))`

File: web/src/routes/auth/change-password/+page.svelte [hunk 9]
```diff
@@ -18,5 +19,12 @@
...
   </p>
 
-  <ChangePasswordForm user={$user} on:success={() => goto(AppRoute.AUTH_LOGIN)} />
+  <ChangePasswordForm
+    user={$user}
+    on:success={async () => {
+      await goto(AppRoute.AUTH_LOGIN);
+      resetSavedUser();
+      await logout();
+    }}
```
Comment (jrasm91) [inline web/src/routes/auth/change-password/+page.svelte:28]: Can you move this on a handleSuccess method defined in the `<script>` section of the file instead?

File: web/src/routes/auth/login/+page.svelte [hunk 10]
```diff
@@ -1,21 +1,12 @@
...
   import type { PageData } from './$types';
 
   export let data: PageData;
-
-  afterNavigate(async ({ from }) => {
-    if (from?.url?.pathname === AppRoute.AUTH_CHANGE_PASSWORD) {
-      resetSavedUser();
-      await logout();
-    }
-  });
```
Comment (jrasm91) [inline web/src/routes/auth/login/+page.svelte:18]: Why did this section get removed?

File: web/svelte.config.js [hunk 11]
```diff
@@ -4,12 +4,6 @@ import { vitePreprocess } from '@sveltejs/vite-plugin-svelte';
 /** @type {import('@sveltejs/kit').Config} */
 const config = {
   preprocess: vitePreprocess(),
-  onwarn: (warning, handler) => {
-    if (warning.code.includes('a11y')) {
-      return;
-    }
-    handler(warning);
-  },
```
Comment (jrasm91) [inline web/svelte.config.js:12]: Is this change related to the async linting?

Comment (benmccann) [reply web/svelte.config.js:12]: Yes, indirectly. I updated SvelteKit to start type-checking the `svelte.config.js` file, which was necessary to get the new eslint rules to work. It didn't like the code here because the types of `warning` and `handler` were not defined. At first I was going to fix it by adding the types here, but then I noticed that when I removed it that I didn't see any warnings printed, so I felt like that was the better fix since it's sort of bad practice to hide the warnings rather than fix them

File: web/src/routes/auth/change-password/+page.svelte (hunk 9 above)
Comment (benmccann) [reply web/src/routes/auth/change-password/+page.svelte]: done. I called it `onSuccess` though. That way when we migrate from event dispatchers to callback props for Svelte 5 we can use the shorthand syntax of `{onSuccess}` rather than having to write `onSuccess={handleSuccess}`

File: web/src/routes/auth/login/+page.svelte (hunk 10 above)
Comment (benmccann) [reply web/src/routes/auth/login/+page.svelte:18]: I moved it to `change-password/+page.svelte` since it was only applicable to that one page

File: web/src/lib/components/asset-viewer/photo-viewer.svelte [hunk 12]
```diff
@@ -120,7 +120,7 @@
     if (state.currentZoom > 1 && isWebCompatibleImage(asset) && !hasZoomed && !$alwaysLoadOriginalFile) {
       hasZoomed = true;
 
-      loadAssetData({ loadOriginal: true });
+      handlePromiseError(loadAssetData({ loadOriginal: true }));
```
[the remark this answers is not in the document]
Comment (benmccann) [reply web/src/lib/components/asset-viewer/photo-viewer.svelte:123]: I can't `await` it because `subscribe` requires a synchronous method

File: web/src/routes/(user)/trash/+page.svelte (hunk 8 above)
Comment (benmccann) [reply web/src/routes/(user)/trash/+page.svelte]: This was one of the very first ones I updated before `handlePromiseError` existed. I probably would have used it if it existed. I've updated it to use it now

File: web/svelte.config.js (hunk 11 above)
Comment (jrasm91) [reply web/svelte.config.js:12]: Sounds good!

**Review by jrasm91 — APPROVED** [review]
Code looks good, but it would be good to do some basic testing as well.

**Comment by benmccann** [comment]
I manually tested the app by navigating around and trying out various pieces of functionality. The only thing I found that looked like a bug also happens on `main`. When you click the heart to favorite or un-favorite a photo, that photo flashes and disappears for a second, which seemed unexpected to me.
